timesketch/lib/sigma_util.py

```diff
@@ -243,3 +247,66 @@ def get_sigma_rule(filepath, sigma_config=None):
             {'file_relpath':file_relpath})
 
         return rule_return
+
+def get_sigma_rule_by_text(rule_text, sigma_config=None):
+    """ Returns a JSON represenation for a rule
+
+    Args:
+        rule_text: Text of the sigma rule to be parsed
+        sigma_config: config file object
+
+    Returns:
+        Json representation of the parsed rule
+    Raises:
+        sigma_exceptions.SigmaParseError: Issue with parsing the given rule
+        yaml.parser.ParserError: Not a correct YAML text provided
+        NotImplementedError: A feature in the provided Sigma rule is not
+            implemented in Sigma for Timesketch
+    """
+    if sigma_config is None:
+        sigma_config = get_sigma_config_file()
+
+    sigma_backend = sigma_es.ElasticsearchQuerystringBackend(sigma_config, {})
+
+    rule_return = {}
+
+    # TODO check if input validation is needed / useful.
+    try:
+        parser = sigma_collection.SigmaCollectionParser(
+            rule_text, sigma_config, None)
+        parsed_sigma_rules = parser.generate(sigma_backend)
+        rule_yaml_data = yaml.safe_load_all(rule_text)
+        for doc in rule_yaml_data:
+            rule_return.update(doc)
+
+    except NotImplementedError as exception:
+        logger.error(
+            'Error generating rule {0!s}'
+            .format(exception))
```

Comment:
nit here nad elsewehere, for consistency in the codebase, keep the `.format` inline with the actual string..

```python
'Error generating rule {0!}'.format(
    exception))
```

that is rather break at the parenthesis instead of at the string end in the case of the format string

Reply:
it actually fitted in one line